### PR TITLE
#297

### LINKED ISSUE #297: Suggestive error message for incorrect review syntax
## Description
500 Internal Server Error is received when the teaching assistant saves a review with incorrect syntax

## How to Reproduce
1. Go to Assignment Review dialog
2. Try to save the review with a first line beginning with:
-0..0: test
3. See error - 500 Server Error

## Expected Behavior
An error message should be sent back to the teaching assistant telling him that the format for the Review is wrong, instead of receiving a 500 error

### PR DESCRIPTION
**Pull request by Lucian-MihaiStan**
## Description
Suggestive error message for incorrect review syntax
Regex fixed for "-0..0: test" and normal tests
Link to syntax format added in "Review" block

Fixes #297 (link the issue)

## Testing
- What was used for testing this PR

## Checklist
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/vmck/acs-interface/labels)
- [x] Tests (if they apply)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Jokeswar) [inline .vscode/settings.json:1]: This comes from VSCode config files. They are not needed in the commit as it "forces" an editor on the developers. Also they can change from one machine to another. I recommend you use a [.gitignore_global](https://docs.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer) to automatically omit the .vscode folder when doing a commit. [Git ignore rules](https://www.atlassian.com/git/tutorials/saving-changes/gitignore).

File: interface/templates/interface/generics/review_modal.html [hunk 2]
```diff
@@ -11,7 +11,14 @@ <h5 class="modal-title" id="reviewLabel">Review submission {{ sub.pk }}</h5>
         <form id="reviewForm" method="post" action="review">
           {% csrf_token %}
           <div class="form-group">
-            <label for="message-text" class="col-form-label">Review details:</label>
+            <div class="row">
+                <div class="col">
+                  <label for="message-text">Review details:</label>
+                </div>
+                <div class="col">
+                  <a href="https://github.com/vmck/acs-interface/wiki/Features#review-submissions" class="float-right" style="text-decoration: none;">Format syntax</a>
```
Comment (Jokeswar) [inline interface/templates/interface/generics/review_modal.html:19]: Can you please supply and image so we can see the result better :smile:.

File: interface/scoring.py [hunk 3]
```diff
@@ -80,7 +80,7 @@ def compute_penalty(
 
 def compute_review_score(submission):
     marks = re.findall(
-        r"^([+-]\d+\.*\d*):",
+        r"^([+-]\d+(?:\.\d+)?):",
```
Comment (Jokeswar) [inline interface/scoring.py:83]: :+1: nice one with the no capture group.

**Review by Jokeswar — CHANGES_REQUESTED** [review]
We are off to a good start :+1:. As a side note, can you please add a more descriptive title to your pull request. I know what the issue #297 is about, but in the future I might not be aware of the issue that the pull request is trying to solve, so a good title might help :smile:.

**Review by Lucian-MihaiStan — COMMENTED** [review]
I will try to add an image/icon so the modications will be much more visible.

File: interface/templates/interface/generics/review_modal.html (hunk 2 above)
Comment (Lucian-MihaiStan) [reply interface/templates/interface/generics/review_modal.html:19]: cropped

**Pull request closed without merging**